jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -215,139 +218,133 @@ private boolean checkImagePlatform(Image image) {
       RegistryClient registryClient, ProgressEventDispatcher progressEventDispatcher)
       throws IOException, RegistryException, LayerPropertyNotFoundException,
           LayerCountMismatchException, BadContainerConfigurationFormatException {
+    Cache cache = buildContext.getBaseImageLayersCache();
+    ImageConfiguration baseImageConfig = buildContext.getBaseImageConfiguration();
+
     ManifestAndDigest<?> manifestAndDigest =
-        registryClient.pullManifest(buildContext.getBaseImageConfiguration().getImageQualifier());
+        registryClient.pullManifest(baseImageConfig.getImageQualifier());
     ManifestTemplate manifestTemplate = manifestAndDigest.getManifest();
 
+    if (manifestTemplate instanceof V21ManifestTemplate) {
+      V21ManifestTemplate v21Manifest = (V21ManifestTemplate) manifestTemplate;
+      cache.writeMetadata(baseImageConfig.getImage(), v21Manifest);
+      return Collections.singletonList(JsonToImageTranslator.toImage(v21Manifest));
```

Comment:
Maybe we should check if the `platform` of the image being returned matches the `platform` specified by the user . We can use the `checkImage`.

Reply:
This is one of the TODO items you're tracking. We should handle it properly, but as a separate PR on its own.

However, note that we shouldn't fail a build even if the platform doesn't match.